README.md

```diff
@@ -1,57 +1,179 @@
 # Setting up Azure IoT Hub File Upload to Azure Storage through Private Endpoint
 
-How to secure Azure IoT Hub's file upload functionality by routing traffic to an Azure Storage account's private endpoint.
+How to enable Azure IoT Hub's file upload functionality through an Azure Storage only allowing private connections.
+
+The file upload functionality provided by Azure IoT Hub serves as a secure bridge to the Azure Storage account, eliminating the need for IoT client devices to rely on external custom services for secure connections to the Storage account. For a comprehensive understanding of this feature, refer to the official  [Azure IoT Hub file upload overview](https://learn.microsoft.com/azure/iot-hub/iot-hub-devguide-file-upload).
+
+This repo offers guidance on securing Azure Storage account through a private endpoint while still allowing IoT Hub to interact with it securely, as well as allowing client IoT devices to reach the account through a gateway (Application Gateway and optionally Azure Firewall). With this approach the Storage account no longer allows public Internet access, which is a common requirement in enterprise deployments.
 
 ## Features
 
-This project framework provides the following features:
+This project provides different options for configuring Azure IoT Hub File upload to a Storage private endpoint:
 
-* Feature 1
-* Feature 2
-* ...
+* [Documentation](./docs/doc-iothub-storage-private.md) with guidance an details on how you can accomplish this configuration in a Hub-Spoke network topology, leveraging Application Gateway, and Azure Firewall for traffic inspection
+* Quickstart: Azure CLI based step-by-step scripts with a simplified setup (single Virtual Network & Resource Group) and sample client app
 
 ## Getting Started
 
 ### Prerequisites
 
-(ideally very short, if any)
+* Azure subscription
```

Comment:
Please include what level of permissions are needed as well on that subscription.